src/sparkx/MultiParticlePtCorrelations.py

```diff
@@ -0,0 +1,500 @@
+                D[order] = (Wk[0]**7. - 21.*Wk[1]*Wk[0]**5.
+                            + 105.*Wk[0]**3.*Wk[1]**2. - 105.*Wk[1]**3.*Wk[0]
+                            + 70.*Wk[2]*Wk[0]**4. - 420.*Wk[2]*Wk[1]*Wk[0]**2.
+                            + 210.*Wk[2]*Wk[1]**2. + 280.*Wk[2]**2.*Wk[0]
+                            - 210.*Wk[3]*Wk[0]**3. - 630.*Wk[3]*Wk[1]*Wk[0]
+                            - 420.*Wk[3]*Wk[2] + 504.*Wk[4]*Wk[0]**2.
+                            - 504.*Wk[4]*Wk[1] - 840.*Wk[5]*Wk[0] + 720.*Wk[6])
+            elif order == 7: # k = 8
+                N[order] = (Pk[0]**8. - 28.*Pk[1]*Pk[0]**6. 
+                            - 210.*Pk[1]**2.*Pk[0]**4. 
+                            - 420.*Pk[1]**3.*Pk[0]**2. + 105.*Pk[1]**4.
+                            + 112.*Pk[2]*Pk[0]**5. + 1120.*Pk[2]*Pk[1]*Pk[0]**3.
+                            + 1680.*Pk[2]*Pk[1]**2.*Pk[0] 
+                            + 1120.*Pk[2]**2.*Pk[0]**2. + 1120.*Pk[2]**2.*Pk[1]
+                            - 420.*Pk[3]*Pk[0]**4. + 2520.*Pk[3]*Pk[1]*Pk[0]**2.
+                            - 1260.*Pk[3]*Pk[1]**2. - 3360.*Pk[3]*Pk[2]*Pk[0]
+                            + 1260.*Pk[4]**2. + 1344.*Pk[4]*Pk[0]**3.
+                            - 4032.*Pk[4]*Pk[1]*Pk[0] + 2688.*Pk[4]*Pk[2]
+                            - 3360.*Pk[5]*Pk[0]**2. + 3360.*Pk[5]*Pk[1]
+                            + 5760.*Pk[6]*Pk[0] - 5040.*Pk[7])
+                D[order] = (Wk[0]**8. - 28.*Wk[1]*Wk[0]**6.
+                            - 210.*Wk[1]**2.*Wk[0]**4.
+                            - 420.*Wk[1]**3.*Wk[0]**2. + 105.*Wk[1]**4.
+                            + 112.*Wk[2]*Wk[0]**5. + 1120.*Wk[2]*Wk[1]*Wk[0]**3.
+                            + 1680.*Wk[2]*Wk[1]**2.*Wk[0]
+                            + 1120.*Wk[2]**2.*Wk[0]**2. + 1120.*Wk[2]**2.*Wk[1]
+                            - 420.*Wk[3]*Wk[0]**4. + 2520.*Wk[3]*Wk[1]*Wk[0]**2.
+                            - 1260.*Wk[3]*Wk[1]**2. - 3360.*Wk[3]*Wk[2]*Wk[0]
+                            + 1260.*Wk[4]**2. + 1344.*Wk[4]*Wk[0]**3.
+                            - 4032.*Wk[4]*Wk[1]*Wk[0] + 2688.*Wk[4]*Wk[2]
+                            - 3360.*Wk[5]*Wk[0]**2. + 3360.*Wk[5]*Wk[1]
+                            + 5760.*Wk[6]*Wk[0] - 5040.*Wk[7])
+
+        self.N_events.append(N)
+        self.D_events.append(D)
+
+    def _compute_numerator_denominator_all_events(self, particle_list_all_events: List[List[Particle]]) -> None:
+        """
+        Compute numerators and denominators of correlations for all events.
+
+        Parameters
+        ----------
+        particle_list_all_events : list
+            List of events, where each event is a list of particle objects.
+        """
+        for event in particle_list_all_events:
+            self._transverse_momentum_correlations_event_num_denom(event)
+
+    def _compute_mean_pt_correlations(self, data: np.ndarray) -> float:
```

Comment:
I think the name data is here a bit unspecific.

Reply:
I changed the name to `numerator_denominator_array`